### PR TITLE
Node.js 10 Upgrade

### PR DESCRIPTION
**Pull request by davestephens**
* Upgrade to Node.js 10
* Upgrade Sqlite3 package to v4 (v3 won't compile with Node.js 10)
* Fix deprecation warnings (all around implementations of Buffers)

I've had a poke around and everything seems OK, but if this can be merged to 0.0.9 I'll get it built into the Docker image and spun up on fORCE9 :+1:

### CODE REVIEW

**Comment by NuSkooler** [comment]
Awesome! This is something I've really wanted to look into - glad you beat me to it! First question: Are there any incompatible issues that y you're aware of with 8.x LTS? (https://github.com/nodejs/Release#release-schedule) Upgrading to the latest shiny is harder for those running ARM systems and the like it seems.

File: core/fnv1a.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ module.exports = class FNV1a {
 
 	digest(encoding) {
 		encoding = encoding || 'binary';
-		let buf = new Buffer(4);
+		let buf = Buffer.alloc(4);
```
Comment (NuSkooler) [inline core/fnv1a.js]: I should have probably been using a const here. FWIW, I think we can use `allocUnsafe()` in cases such as this where we *know* we're about to overwrite the entire chunk of memory anyway (https://nodejs.org/api/buffer.html#buffer_buffer_from_buffer_alloc_and_buffer_allocunsafe)

File: package.json [hunk 2]
```diff
@@ -56,6 +56,6 @@
   },
   "devDependencies": {},
   "engines": {
-    "node": ">=6.9.2"
+    "node": ">=10"
```
Comment (NuSkooler) [inline package.json]: RE: the previous question, I'm *hoping* we can keep this 8.x LTS+... I'm just not sure yet.

**Comment by davestephens** [comment]
I've pushed a few fixes - I knew I'd miss one or two allocs vs froms! 

I left the allocs as allocs rather than allocUnsafe as I figured the speed difference would be negligible, and the flip-side would likely be one of those bugs that goes unnoticed until something really weird happens at some point down the line due to memory not being cleared before being used  ;-)

RE Node.js 10 - I need to test it on my Pi3. It's the LTS-to-be (as I'm sure you're aware) - so figured it wouldn't hurt jumping the gun in setting it as a requirement - assuming it works everywhere of course!!

**Comment by NuSkooler** [comment]
@davestephens Sweet, I'll try out this branch with the latest 8.x and see how it goes as well. It would be a bonus if it worked, or if it was trivial to make work with both, but if not I agree moving to 10.x is still the way to go

File: core/archive_util.js [hunk 3]
```diff
@@ -78,11 +78,11 @@ module.exports = class ArchiveUtil {
 			Object.keys(Config.fileTypes).forEach(mimeType => {
 				const fileType = Config.fileTypes[mimeType];
 				if(fileType.sig) {
-					fileType.sig 	= new Buffer(fileType.sig, 'hex');
+					fileType.sig 	= Buffer.alloc(fileType.sig, 'hex');
```
Comment (NuSkooler) [inline core/archive_util.js]: This one needs to use `.from()`

**Comment by davestephens** [comment]
OK, I think we're there...

**Comment by NuSkooler** [comment]
Wahoo!

**Comment by davestephens** [comment]
Wah wah we wah!

**Pull request merged**